### DISCUSSION TITLE
[8.0 QA] Explicit Actions Args Mocking

### DISCUSSION THREAD

**Discussion by vanessayuenn** [storybookjs/storybook#25690] [Announcement]
#### Feature owner: @kasperpeulen 
Please comment in this discussion thread for any bug report and feedback for this feature as you test out Storybook 8. Kindly adhere to the following format:

> **Project**: Name of your project as entered in Notion
> **Version**: The version of storybook you tested with
> **Bug/feedback**: Details of the bugs/feedback you have

Feature owner should triage the comments here frequently and open issues as needed.

**Comment by yannbf** [comment]
**Project:** MealDrop
**Version:** 8.0.0-alpha.10
**Bug/feedback:** 

When upgrading Storybook and running tests using portable stories (React), there's an error: `Object.defineProperty called on non-object`:
<img src="https://github.com/storybookjs/storybook/assets/1671563/a44d97de-6f43-49c4-8e63-2443349a39a0" width="400" />

The issue seems to happen when `composeStories` is called on a story file that invokes anything from `@storybook/test`.
This issue was introduced in [alpha 10](https://github.com/storybookjs/storybook/releases/tag/v8.0.0-alpha.10), potentially from [this PR](https://github.com/storybookjs/storybook/pull/25576)

Reproduction:
1. Access https://github.com/yannbf/mealdrop/tree/bug/portable-stories-expect
2. Run `yarn install` and `yarn test`

Here are the two places to focus on:
- [Test file](https://github.com/yannbf/mealdrop/blob/bug/portable-stories-expect/src/pages/RestaurantDetailPage/RestaurantDetailPage.test.tsx#L7)
- [Story file](https://github.com/yannbf/mealdrop/blob/bug/portable-stories-expect/src/pages/RestaurantDetailPage/RestaurantDetailPage.stories.tsx#L51)

**Reply by vanessayuenn** [reply]
tracked in https://github.com/storybookjs/storybook/issues/25794

**Comment by shilman** [comment]
**Project** Storyblok
**Version** 8.0.0-alpha.15

I added the following story:

```js
import { userEvent, within, expect } from '@storybook/test'
import SbButton from './SbButton.vue'

export default {
  component: SbButton,
  argTypes: {
    handleClick: { type: 'function' },
  },
  parameters: {
    actions: { argTypesRegex: '^handle[A-Z].*' },
  },
}

export const Default = {
  args: {
    label: 'Hmmhahahah',
    size: 'large',
  },
}

export const Click = {
  ...Default,
  play: async (context: any): Promise<void> => {
    const { canvasElement } = context
    const canvas = within(canvasElement)
    await userEvent.click(canvas.getByRole('button'))
    expect(context.args.handleClick).toHaveBeenCalled()
  },
}

export const ClickMocked = {
  ...Click,
  args: {
    ...Click.args,
    handleClick: fn(),
  },
}
```

It gives the following error in both the UI and the browser console, which seems incorrect:

<img width="782" alt="image" src="https://github.com/storybookjs/storybook/assets/488689/f2ff013e-ab38-47bf-8ee2-83f52d3bb8ca">

**Reply by shilman** [reply]
The `ClickMocked` also doesn't work, but I suspect it's something to do with the args not getting passed correctly as events to the Vue component. @kasperpeulen do you know the fix here off the top of your head?

**Reply by vanessayuenn** [reply]
tracked https://github.com/storybookjs/storybook/issues/25810
